src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -867,28 +644,75 @@ public static String swapToolGrade(final int toolGrade)
     /**
      * Adapted from {@link net.minecraft.entity.player.InventoryPlayer#addItemStackToInventory(ItemStack)}.
      *
-     * @param inventory Inventory to add itemstack to.
+     * @param handler Inventory to add itemstack to.
      * @param itemStack ItemStack to add.
      * @param building the building.
      * @return itemStack which has been replaced.
      */
     @Nullable
-    public static ItemStack forceItemStackToInventory(@NotNull final IInventory inventory, @NotNull final ItemStack itemStack, @NotNull final AbstractBuilding building)
+    public static ItemStack forceItemStackToInventory(@NotNull final IItemHandler handler, @NotNull final ItemStack itemStack, @NotNull final AbstractBuilding building)
     {
-        if(!addItemStackToInventory(inventory, itemStack))
+        if(!addItemStackToInventory(handler, itemStack))
         {
             final List<ItemStorage> localAlreadyKept = new ArrayList<>();
-            for(int i = 0; i < inventory.getSizeInventory(); i++)
+            for(int i = 0; i < handler.getSlots(); i++)
             {
-                final ItemStack localStack = inventory.getStackInSlot(i);
+                final ItemStack localStack = handler.getStackInSlot(i);
                 if(!EntityAIWorkDeliveryman.workerRequiresItem(building, localStack, localAlreadyKept))
                 {
-                    final ItemStack removedStack = inventory.removeStackFromSlot(i);
-                    inventory.setInventorySlotContents(i, itemStack.copy());
+                    final ItemStack removedStack = handler.extractItem(i, localStack.stackSize, false);
+                    if (handler.insertItem(i, itemStack.copy(), false) != null)
+                    {
+                        MineColonies.getLogger().error("forceItemStackToInventory failed forcing! This is a bug - please report to MineColonies developers!");
+                    }
                     return removedStack.copy();
                 }
             }
         }
         return null;
     }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)